src/systems/mouse_ray.rs

```diff
@@ -0,0 +1,78 @@
+use amethyst::core::cgmath::prelude::*;
+use amethyst::core::cgmath::{Matrix4, Point3, Vector3, Vector4};
+use amethyst::core::specs::prelude::*;
+use amethyst::core::transform::Transform;
+use amethyst::input::InputHandler;
+use amethyst::renderer::ActiveCamera;
+use amethyst::renderer::{Camera, ScreenDimensions};
+use collision::Ray3;
+
+pub struct MouseRaySystem;
+
+impl<'s> System<'s> for MouseRaySystem {
+    type SystemData = (
+        Read<'s, InputHandler<String, String>>,
+        Option<ReadExpect<'s, ActiveCamera>>,
+        ReadExpect<'s, ScreenDimensions>,
```

Comment:
can you get the dimensions of the 3d renderer instead? that would make more sense semantically.